src/ring_theory/witt_vector/isocrystal.lean

```diff
@@ -0,0 +1,201 @@
+/-
+Copyright (c) 2022 Heather Macbeth. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Heather Macbeth
+-/
+
+import ring_theory.witt_vector.is_alg_closed
+
+/-!
+
+## F-isocrystals over a perfect field
+
+When `k` is an integral domain, so is `𝕎 k`, and we can consider its field of fractions `K(p, k)`.
+The endomorphism `witt_vector.frobenius` lifts to `φ : K(p, k) → K(p, k)`; if `k` is perfect, `φ` is
+an automorphism.
+
+Let `k` be an algebraically closed field of characteristic `p`. It is thus a perfect integral
+domain. Let `V` be a vector space over `k`. An *isocrystal* is a bijective map `V → V` that is
```

Comment:
```suggestion
Let `k` be a perfect integral domain. Let `V` be a vector space over `k`.
An *isocrystal* is a bijective map `V → V` that is
```
I believe we don't assume `k` to be an algebraically closed field until the final theorem.